osquery/tables/system/windows/windows_security_center.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <windows.h>
+#include <wscapi.h>
+
+#include <osquery/core.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+#include <osquery/utils/conversions/windows/strings.h>
+#include <osquery/utils/map_take.h>
+
+namespace osquery {
+namespace tables {
+
+const auto kSecurityProviderStates = std::unordered_map<int, std::string>{
+    {WSC_SECURITY_PROVIDER_HEALTH_NOTMONITORED, "Not Monitored"},
+    {WSC_SECURITY_PROVIDER_HEALTH_GOOD, "Good"},
+    {WSC_SECURITY_PROVIDER_HEALTH_POOR, "Poor"},
+    {WSC_SECURITY_PROVIDER_HEALTH_SNOOZE, "Snoozed"},
+};
+
+std::string resolveProductHealthOrError(int productName) {
+  // Attempt a runtime link to the DLL containing these functions,
+  // since linking the library was causing a crash on some Windows
+  // machines (like the CI server).
+  typedef HRESULT(WINAPI * pWscGetSecurityProviderHealth)(
+      _In_ DWORD Providers, _Out_ PWSC_SECURITY_PROVIDER_HEALTH);
+  pWscGetSecurityProviderHealth WscGetSecurityProviderHealth;
+  HMODULE hDLL = LoadLibrary(TEXT("wscapi.dll"));
+
+  if (hDLL) {
+    HRESULT result = E_UNEXPECTED;
+    WSC_SECURITY_PROVIDER_HEALTH health;
+    WscGetSecurityProviderHealth =
+        (pWscGetSecurityProviderHealth)GetProcAddress(
+            hDLL, "WscGetSecurityProviderHealth");
+    if (WscGetSecurityProviderHealth != NULL) {
+      result = WscGetSecurityProviderHealth(productName, &health);
+      if (result == S_OK) {
+        return tryTakeCopy(kSecurityProviderStates, health)
+            .takeOr(std::string("Unknown"));
+      }
+      return "Error";
+    }
+  }
+
+  return "Error";
+}
+
+bool windowsUpdateServicesEnabled() {
+  auto wuauservResult =
+      SQL::selectAllFrom("services", "name", EQUALS, "wuauserv");
+  auto usosvcResult = SQL::selectAllFrom("services", "name", EQUALS, "UsoSvc");
+
+  if (wuauservResult.empty() || usosvcResult.empty()) {
+    return false;
+  }
+
+  auto& wuauservRow = wuauservResult.at(0);
+  auto& usosvcRow = usosvcResult.at(0);
+
+  if (wuauservRow.at("start_type") == "DISABLED" ||
```

Comment:
Check `.count()` before using `.at()` to avoid possible exception (for example if this column name was ever changed or dropped).